samples/synchronization/src/main.c

```diff
@@ -97,9 +97,13 @@ void threadA(void *dummy1, void *dummy2, void *dummy3)
 	/* spawn threadB */
 	k_tid_t tid = k_thread_create(&threadB_data, threadB_stack_area,
 			STACKSIZE, threadB, NULL, NULL, NULL,
-			PRIORITY, 0, K_NO_WAIT);
+			PRIORITY, 0, K_FOREVER);
 
 	k_thread_name_set(tid, "thread_b");
+#if CONFIG_SCHED_CPU_MASK
+	k_thread_cpu_mask_enable(&threadB_data, 0);
```

Comment:
The default state is all CPU's enabled, so you want to call k_thread_cpu_disable_all() first.  By itself this is a noop on a brand new thread.

Reply:
ah ok, let me try that...